File: 10_week_2/tribonacci.js
var tribonacci = [0, 0, 1];

function tricky_tribonacci(num) {
	if (num < 0) {
		return null;
	}

	while (tribonacci.length <= num) {
		var previous2 = tribonacci[tribonacci.length - 3];
		var previous1 = tribonacci[tribonacci.length - 2];
		var current = tribonacci[tribonacci.length - 1];

		tribonacci.push(previous2 + previous1 + current);
	}

	return tribonacci[num];
}

console.log(tricky_tribonacci(9));

File: 10_week_2/tribonacci.py
tribonacci = [0, 0, 1]

def tricky_tribonacci(num):
	if num < 0:
		return None

	while len(tribonacci) <= num:
		previous2 = tribonacci[len(tribonacci) - 3]
		previous1 = tribonacci[len(tribonacci) - 2]
		current = tribonacci[len(tribonacci) - 1]

		tribonacci.append(previous2 + previous1 + current)

	return tribonacci[num]

print(tricky_tribonacci(9))
